crates/solver/src/settlement_submission/submitter/flashbots_api.rs

```diff
@@ -25,7 +28,11 @@ impl TransactionSubmitting for FlashbotsApi {
         super::common::submit_raw_transaction(self.client.clone(), URL, tx).await
     }
 
-    async fn cancel_transaction(&self, _id: &TransactionHandle) -> Result<()> {
+    async fn cancel_transaction(&self, _id: &CancelHandle) -> Result<()> {
+        Ok(())
+    }
```

Comment:
Canceling on flashbots does nothing because our transaction is guaranteed to be included and therefore we never have to cancel, right?
Should this be documented or asserted?

Reply:
See thread above for the explanation.

Reply:
Flashbots do provide cancelling logic, but two things could happen:
1. Our tx is sitting in flashbots mempool and its not forwarded to miners (still) -> our cancel removes tx from flashbots mempool
2. Our tx is forwarded to miners -> our cancel only removes tx from flashbots mempool but does not guarantee that our tx wont be mined (because there is a chance that miners mined a block containing our tx in the time between our cancel request and miners receiving a new block proposal not containing our tx)

On our side, cancelling on Flashbots is not that critical, because once we detect the tx will fail, our submission logic ends, and it starts running a new loop with a new settlement. In that case, our tx wont be mined as failed, because Flashbots gives that guarantee.